core/src/main/java/org/jruby/RubyMatchData.java

```diff
@@ -846,27 +848,47 @@ public RubyFixnum hash() {
         return metaClass.runtime.newFixnum( hashCode() );
     }
 
-    @JRubyMethod
-    public RubyHash named_captures(ThreadContext context) {
+    @JRubyMethod(keywords = true, checkArity = false, optional = 1)
+    public RubyHash named_captures(ThreadContext context, IRubyObject[] args) {
         check();
+
         Ruby runtime = context.runtime;
+        int argc = Arity.checkArgumentCount(runtime, args.length, 0, 0, true);
         RubyHash hash = RubyHash.newHash(runtime);
         if (regexp == context.nil) return hash;
 
+        IRubyObject symbolizeNamesValue = runtime.getNil();
```

Comment:
I think this is probably better as a boolean and lower perform the isTrue when processing the option.